src/dirext.rs

```diff
@@ -215,6 +215,16 @@ pub trait CapStdExtDirExtUtf8 {
         contents: impl AsRef<[u8]>,
         perms: cap_std::fs::Permissions,
     ) -> Result<()>;
+
+    /// Read all filenames in this directory
+    fn filenames(&self) -> Result<Vec<String>> {
```

Comment:
I would intuitively expect these sorts of methods to return iterators, not vecs, so the caller could do iterator-y things with them if desired.  I guess something like `-> impl Iterator<Item = Result<String>>` since `DirEntry::file_name()` can `Err`.

Reply:
You're right, but I see this as a high level convenience helper API for those who literally just want the filenames as we've seen in a few places, especially to *sort* them - that's where you need to collect into a vector.

When I was writing this I actually added `filenames_sorted_by()` (and a thin wrapper `filenames_sorted()`...so we could jump up to that level directly too...?)

Reply:
Yeah I think that would give the best of both worlds.  The `sorted(_by)` methods I would expect to give back the `Vec`, then the others would give `Iterator`.

Reply:
OK I made this one for now *just* about sorting, which always wants a `Vec`.

For the use cases that want iteration it's not clear to me we can do much more convenient than what exists today with

```
for ent in dir.entries()? {
   let ent = ent?;
   let name = ent.file_name()?;
}
```